torch/csrc/jit/script/parser.h

```diff
@@ -182,11 +185,58 @@ struct Parser {
       L.expect(end);
     return List<T>::create(r, elements);
   }
+
   Const parseConst() {
     auto range = L.cur().range;
     auto t = L.expect(TK_NUMBER);
     return Const::create(t.range, t.text());
   }
+
+  bool isCharCount(char c, const std::string& str, size_t start, int len) {
+    //count checks from [start, start + len) 
+    return start + len <= str.size() && std::count(str.begin() + start, str.begin() + start + len, c) == len; 
+  }
+
+  std::string parseLexedString(const std::string &str, size_t start) {
+    //invariants: input consists of a valid adjacent python strings and start is the index at beginning 
+    //of a string
+    char quote = str[start];
+    int quote_len = isCharCount(quote, str, start, 3) ? 3 : 1;
+    size_t end = start + quote_len;
+    while(!isCharCount(quote, str, end, quote_len)) {
+      if (str[end] == '\\') {
+        end++;
+      }
+      end++;
+    }
+    std::string ret_str = str.substr(start + quote_len, /*len*/ end - (start + quote_len));
+    //erase line continuations
+    std::string sub = "\\\n";
+    size_t pos = ret_str.find(sub);
+    while(pos != std::string::npos) {   
+        ret_str.erase(pos, sub.size());
+        pos = ret_str.find(sub);
+    }
+    size_t next_start = end + quote_len; 
+    //find start of next string
+    while(next_start < str.size() && str[next_start] != '\'' && str[next_start] != '\"') {
+      if (str[next_start] == '#') {
+        next_start = str.find('\n', next_start);
+      }
+      next_start++;
+    }
```

Comment:
Escaped characters are never handled. '\n' will turn into '\n' rather than a newline, for instance. It is sufficient for this PR to only handle newlines.

Reply:
The only escaped characters that have to be handled are ones that python print and c++ print differently right ?